opae.spec.fedora

```diff
@@ -0,0 +1,328 @@
+
+
+BuildRequires:  make
+BuildRequires:  cmake
+BuildRequires:  git
+BuildRequires:  gcc
+BuildRequires:  gcc-c++
+BuildRequires:  json-c-devel
+BuildRequires:  libuuid-devel
+BuildRequires:  python3-devel
+BuildRequires:  python3-jsonschema
+BuildRequires:  python3-pip
+BuildRequires:  python3-setuptools
+BuildRequires:  rpm-build
+BuildRequires:  systemd-devel
+BuildRequires:  libcap-devel
+BuildRequires:  tbb-devel
+BuildRequires:  hwloc-devel
+BuildRequires:  pybind11-devel
+BuildRequires:  cli11-devel
+BuildRequires:  spdlog-devel
+BuildRequires:  libedit-devel
+
+
+%description
+Open Programmable Acceleration Engine (OPAE) is a software framework
+for managing and accessing programmable accelerators (FPGAs).
+Its main parts are:
+
+* OPAE Software Development Kit (OPAE SDK) (this package)
+* OPAE Linux driver for Intel(R) Xeon(R) CPU with
+  Integrated FPGAs and Intel(R) PAC with Arria(R) 10 GX FPGA
+* Basic Building Block (BBB) library for accelerating AFU
+
+OPAE SDK is a collection of libraries and tools to facilitate the
+development of software applications and accelerators using OPAE.
+It provides a library implementing the OPAE C API for presenting a
+streamlined and easy-to-use interface for software applications to
+discover, access, and manage FPGA devices and accelerators using
+the OPAE software stack.
+
+%package devel
+Summary:    OPAE headers, sample source, and documentation
+Requires:   libuuid-devel, %{name}%{?_isa} = %{version}-%{release}
+
+%description devel
+OPAE headers, tools, sample source, and documentation
+
+
+%{?python_disable_dependency_generator}
+# Workaround a problem with pybind11 *.so not having build-id's
+%undefine _missing_build_ids_terminate_build
+
+
+%prep
+%setup -q -n %{name}-%{version}-%{opae_release}
+
+%build
+%cmake -DCMAKE_INSTALL_PREFIX=/usr \
+       -DOPAE_PRESERVE_REPOS=ON \
+       -DOPAE_BUILD_LEGACY=ON \
+       -DOPAE_BUILD_EXTRA_TOOLS_FPGABIST=ON \
+       -DOPAE_BUILD_EXTRA_TOOLS_OPAEIO=ON \
+       -DOPAE_WITH_CLI11=ON \
+       -DOPAE_WITH_SPDLOG=ON \
+       -DOPAE_WITH_PYBIND11=ON .
+%if 0%{?rhel}
+  %make_build
+%else
+  %cmake_build
+%endif
+
+
+%install
+mkdir -p %{buildroot}%{_datadir}/opae
+cp RELEASE_NOTES.md %{buildroot}%{_datadir}/opae/RELEASE_NOTES.md
+cp LICENSE %{buildroot}%{_datadir}/opae/LICENSE
+cp COPYING %{buildroot}%{_datadir}/opae/COPYING
+
+mkdir -p %{buildroot}%{_usr}/src/opae/cmake/modules
+
+mkdir -p %{buildroot}%{_usr}/src/opae/opae-libs/cmake/modules
+for s in FindHwloc.cmake \
+         OPAE.cmake \
+         FindUUID.cmake \
+         FindCap.cmake \
+         FindUdev.cmake \
+         Findjson-c.cmake \
+         OPAECompiler.cmake \
+         OPAEGit.cmake \
+         OPAEPackaging.cmake 
+do
+  cp "opae-libs/cmake/modules/${s}" %{buildroot}%{_usr}/src/opae/opae-libs/cmake/modules
+  chmod a+x %{buildroot}%{_usr}/src/opae/opae-libs/cmake/modules/$s
+done
+
+mkdir -p %{buildroot}%{_usr}/src/opae/samples
```

Comment:
please  remove samples folder  and scource code  from spec file
Fedora rom spec create SRC rpm